### PR TITLE
feat: allow Offer to accept a PaymentPKeywordRecord

### PR DESCRIPTION
**Pull request by michaelfig**
This change was requested by @dtribble.  Please review.

### CODE REVIEW

File: packages/zoe/src/zoeService/zoe.js [hunk 1]
```diff
@@ -380,7 +380,9 @@ function makeZoe(vatAdminSvc, zcfBundleName = undefined) {
             // cleaned proposal's amount that should be the same.
             const giveAmount = proposal.give[keyword];
             const purse = brandToPurse.get(giveAmount.brand);
-            return E(purse).deposit(paymentKeywordRecord[keyword], giveAmount);
+            return E.when(paymentKeywordRecord[keyword], payment =>
+              E(purse).deposit(payment, giveAmount),
```
Comment (erights) [inline packages/zoe/src/zoeService/zoe.js:384]: Interesting that even I was confused at first about why you're doing an `E.when` here. Would your types cause the IDE to flag the pre-PR line
```js
return E(purse).deposit(paymentKeywordRecord[keyword], giveAmount);
```
as being a type violation given the post-PR repaired type declarations? If so, that's awesome, as this is an easy mistake to make.

**Review by erights — APPROVED** [review]
LGTM

File: packages/zoe/src/zoeService/zoe.js (hunk 1 above)
Comment (michaelfig) [reply packages/zoe/src/zoeService/zoe.js:384]: Yes, caught by the type system.

**Comment by katelynsills** [comment]
Thanks!

**Pull request merged**